invisible_cities/core/fit_functions.py

```diff
@@ -96,16 +125,26 @@ def fit(func, x, y, seed=(), fit_range=None, **kwargs):
     vals, cov = scipy.optimize.curve_fit(func,
                                          x, y,
                                          seed,
+                                         absolute_sigma=True,
                                          **kwargs)
 
     fitf = lambda x: func(x, *vals)
     fitx = fitf(x)
-    chi2, pval = scipy.stats.chisquare(y[fitx != 0], fitx[fitx != 0], len(vals))
+
```

Comment:
use a variable to hold the result of `fitx != 0`. It is used in 4 different places.